### PR TITLE
infowindows: json array

### PR DESCRIPTION
**Pull request by matallo**
close https://github.com/CartoDB/cartodb/issues/9171

replicating the implementation of current infowindows, which support json arrays, right now we can now a infowindow is custom if it has `template` but `template_name` is empty

screen shot 2016-08-18 at 18 18 11

CR @xavijam @javisantana 

/cc @ramiroaznar

### CODE REVIEW

**Comment by matallo** [comment]
damn, fixing those tests :/

File: src/geo/ui/infowindow-view.js [hunk 1]
```diff
@@ -107,8 +111,12 @@ var Infowindow = View.extend({
 
       var data = this.model.get('content') ? this.model.get('content').data : {};
 
-      // Sanitized fields
-      fields = _.map(fields, this._sanitizeField, this);
+      // If a custom template is not applied, let's sanitize
+      // fields for the template rendering
+      if (!this.isCustomTemplate()) {
```
Comment (alonsogarciapablo) [inline src/geo/ui/infowindow-view.js]: Aren't all templates "custom" in the context of cartodb.js now that we don't rely on `template_name` anymore?

**Comment by matallo** [comment]
indeed they are, but these still work with _old_ templates where only template name (and no template attribute) is given

**Comment by matallo** [comment]
aaand tests passing again

**Comment by xavijam** [comment]
I can't add much more, all agree?

**Comment by alonsogarciapablo** [comment]
Talked to @matallo about this solution. While I think it solves the problem we're currently seeing with old editor infowindow templates (CartoDB/cartodb#9171), I think we can fix this in a more generic way. Users might encounter this problem when using the API.

We found out that we're [sanitising fields here](https://github.com/CartoDB/cartodb.js/blob/666d671e006bff8faa10ecedf1113b738d4f1f79/src/geo/ui/infowindow-view.js#L275) and this is not working properly when a field has an object as it's value... My suggestion would be to try to handle this case gracefully instead of skipping sanitisation when something that the builder sets (today) (`template_name`) in some cartodb.js models is present.

@matallo I can help you with this, ofc! Thanks and... avant!

**Comment by matallo** [comment]
@alonsogarciapablo finally, sorry it took too much (holidays, priorities, ...)

- added corresponding tests

/cc @ernesmb

File: src/geo/ui/infowindow-view.js [hunk 2]
```diff
@@ -246,6 +246,16 @@ var Infowindow = View.extend({
     }
   },
 
+  _sanitizeValue: function (key, obj) {
+    var t = typeof (obj);
+
+    if (t !== 'object' || obj === null) {
```
Comment (alonsogarciapablo) [inline src/geo/ui/infowindow-view.js]: Suggestion: we could use [`_.isObject`](http://underscorejs.org/#isObject) and [`isArray`](http://underscorejs.org/#isArray) here... 😄

Comment (matallo) [inline src/geo/ui/infowindow-view.js]: ooh, totally, changing and publishing! ^^

**Pull request merged**